component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java

```diff
@@ -149,14 +171,26 @@ private PipelineResult runDataReadingPipeline() {
 
             try {
                 Thread.currentThread().setContextClassLoader(beamAwareClassLoader);
-
-                PipelineOptions options = PipelineOptionsFactory.create();
-                PushRecord pushRecord = new PushRecord();
+                DirectOptions options = PipelineOptionsFactory.as(DirectOptions.class);
+                options.setRunner(DirectRunner.class);
+                options.setTargetParallelism(10);
```

Comment:
Isn't that big? I don't know the implications on this parameter...

Reply:
May be too big indeed (should be less than QUEUE size), this is the number of sub-thread of Beam thread